scripts/usb_reset.py

```diff
@@ -212,17 +212,17 @@ def setup_cgroup(domid, pid):
 
     try:
         # unbuffered write to ensure each one is flushed immediately
-        with open(cg_dir + "/tasks", "w", 0) as tasks, \
-                open(cg_dir + "/devices.deny", "w", 0) as deny, \
-                open(cg_dir + "/devices.allow", "w", 0) as allow:
+        with open(cg_dir + "/tasks", "wb") as tasks, \
+                open(cg_dir + "/devices.deny", "wb") as deny, \
+                open(cg_dir + "/devices.allow", "wb") as allow:
```

Comment:
Due to cgroupfs and the comment above it, keep using unbuffered I/O and explain why switching to binary I/O (the written bytes are the same):
```suggestion
        # Writing to cgroupfs files is done unbuffered to ensure the order of writes.
        # Unbuffered I/O needs binary I/O in Python3:
        with open(cg_dir + "/tasks", "wb", 0) as tasks, \
                open(cg_dir + "/devices.deny", "wb", 0) as deny, \
                open(cg_dir + "/devices.allow", "wb", 0) as allow:
```